src/test/java/org/isf/admission/Tests.java

```diff
@@ -1078,6 +1141,193 @@ void testMgrDeletePatientPhoto(boolean maternityRestartInJune) throws Exception
+		assertThatThrownBy(() -> admissionBrowserManager.newAdmission(admission))
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								e -> ((OHServiceException) e).getMessages().size() == 1, "Expecting single validation error"));
+	}
+
+	@ParameterizedTest(name = "Test with MATERNITYRESTARTINJUNE={0}")
+	@MethodSource("maternityRestartInJune")
+	void testMgrValidateDiseaseIn(boolean maternityRestartInJune) throws Exception {
+		GeneralData.MATERNITYRESTARTINJUNE = maternityRestartInJune;
+		int id = setupTestAdmission(false);
+		Admission admission = admissionBrowserManager.getAdmission(id);
+		GeneralData.LANGUAGE = "en";
+
+		// DiseaseIn cannot be null
+		admission.setDiseaseIn(null);
+		assertThatThrownBy(() -> admissionBrowserManager.updateAdmission(admission))
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								e -> ((OHServiceException) e).getMessages().size() == 1, "Expecting single validation error"));
+	}
+
+	@ParameterizedTest(name = "Test with MATERNITYRESTARTINJUNE={0}")
+	@MethodSource("maternityRestartInJune")
+	void testMgrValidateWeight(boolean maternityRestartInJune) throws Exception {
+		GeneralData.MATERNITYRESTARTINJUNE = maternityRestartInJune;
+		int id = setupTestAdmission(false);
+		Admission admission = admissionBrowserManager.getAdmission(id);
+		GeneralData.LANGUAGE = "en";
+
+		// Weight cannot be negative
+		admission.setWeight(-99.0f);
+		assertThatThrownBy(() -> admissionBrowserManager.updateAdmission(admission))
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								e -> ((OHServiceException) e).getMessages().size() == 1, "Expecting single validation error"));
+	}
+
+	@ParameterizedTest(name = "Test with MATERNITYRESTARTINJUNE={0}")
+	@MethodSource("maternityRestartInJune")
+	void testMgrValidateMaternityVisitDate(boolean maternityRestartInJune) throws Exception {
+		GeneralData.MATERNITYRESTARTINJUNE = maternityRestartInJune;
+		int id = setupTestAdmission(false, true);
+		Admission admission = admissionBrowserManager.getAdmission(id);
+		GeneralData.LANGUAGE = "en";
+
+		admission.setVisitDate(admission.getAdmDate().minusDays(30));
+		assertThatThrownBy(() -> admissionBrowserManager.updateAdmission(admission))
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								e -> ((OHServiceException) e).getMessages().size() == 1, "Expecting single validation error"));
+	}
+
+	@ParameterizedTest(name = "Test with MATERNITYRESTARTINJUNE={0}")
+	@MethodSource("maternityRestartInJune")
+	void testMgrValidateDeliveryDate(boolean maternityRestartInJune) throws Exception {
+		GeneralData.MATERNITYRESTARTINJUNE = maternityRestartInJune;
+		int id = setupTestAdmission(false, true);
+		Admission admission = admissionBrowserManager.getAdmission(id);
+		GeneralData.LANGUAGE = "en";
+
+		admission.setVisitDate(null);
```

Comment:
this should not be needed for this test

Reply:
Yes it is required to force a different code path; I was trying to execute both branches of a test.

Reply:
So I believe this should stay.

Reply:
Ah ok, but it doesn't change the test outcome, does it?

Reply:
no; it just increases line coverage 😄